sdk/communication/azure-communication-chat/azure/communication/chat/_shared/models.py

```diff
@@ -173,19 +198,81 @@ def __init__(self, user_id, **kwargs):
         if self.raw_id is None:
             self.raw_id = _microsoft_teams_user_raw_id(self)
 
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
+
 
 def _microsoft_teams_user_raw_id(identifier: MicrosoftTeamsUserIdentifier) -> str:
     user_id = identifier.properties['user_id']
     if identifier.properties['is_anonymous']:
-        return '8:teamsvisitor:{}'.format(user_id)
+        return f'{TEAMS_USER_ANONYMOUS}{user_id}'
     cloud = identifier.properties['cloud']
     if cloud == CommunicationCloudEnvironment.DOD:
-        return '8:dod:{}'.format(user_id)
+        return f'{TEAMS_USER_DOD_CLOUD}{user_id}'
     elif cloud == CommunicationCloudEnvironment.GCCH:
-        return '8:gcch:{}'.format(user_id)
+        return f'{TEAMS_USER_GCCH_CLOUD}{user_id}'
     elif cloud == CommunicationCloudEnvironment.PUBLIC:
-        return '8:orgid:{}'.format(user_id)
-    return '8:orgid:{}'.format(user_id)
+        return f'{TEAMS_USER_PUBLIC_CLOUD}{user_id}'
+    return f'{TEAMS_USER_PUBLIC_CLOUD}{user_id}'
+
+
+MicrosoftBotProperties = TypedDict(
+    'MicrosoftBotProperties',
+    bot_id=str,
+    is_global=bool,
+    cloud=Union[CommunicationCloudEnvironment, str]
+)
+
+
+class MicrosoftBotIdentifier(object):
+    """Represents an identifier for a Microsoft bot.
+
+    :ivar str raw_id: Optional raw ID of the identifier.
+    :ivar kind: The type of identifier.
+    :vartype kind: str or CommunicationIdentifierKind
+    :ivar Mapping properties: The properties of the identifier.
+     The keys in this mapping include:
+        - `bot_id`(str): The id of the Microsoft bot.
+        - `is_global` (bool): Set this to true if the bot is global.
+        - `cloud` (str): Cloud environment that this identifier belongs to.
+
+    :param str bot_id: Microsoft bot id.
+    :keyword bool is_global: `True` if the identifier is global. Default value is `False`.
+    :keyword cloud: Cloud environment that the bot belongs to. Default value is `PUBLIC`.
+    :paramtype cloud: str or ~azure.communication.chat.CommunicationCloudEnvironment
+    """
+    kind = CommunicationIdentifierKind.MICROSOFT_BOT
+
+    def __init__(self, bot_id, **kwargs):
+        # type: (str, Any) -> None
+        self.raw_id = kwargs.get('raw_id')
+        self.properties = MicrosoftBotProperties(
+            bot_id=bot_id,
+            is_global=kwargs.get('is_global', False),
```

Comment:
can we add tests for default value scenarios pls

Reply:
could not find the tests, I was expecting the following tests:
[microsoftBotIdentifier_defaultCloudIsPublic](https://github.com/Azure/azure-sdk-for-android/blob/60733f70961b66fc4921262305b4a065e2086ffe/sdk/communication/azure-communication-common/src/test/java/com/azure/android/communication/common/CommunicationIdentifierTests.java#L33)
[microsoftBotIdentifier_defaultGlobalIsFalse](https://github.com/Azure/azure-sdk-for-android/blob/60733f70961b66fc4921262305b4a065e2086ffe/sdk/communication/azure-communication-common/src/test/java/com/azure/android/communication/common/CommunicationIdentifierTests.java#L45)

Reply:
@AikoBB I see now what you wanted to say.
I added tests here:
https://github.com/Azure/azure-sdk-for-python/pull/28910/commits/3255fa06e2689f7850c22a8e18d5f29271d5d1c4#diff-179ac094128a26fa8e212a40f4f3fe476a57c709f42d3a75322b8851f34806c5R511-R533

The last one also covers the case of default isGlobal = false flag

Reply:
in these tests, you are asserting if the two objects are equal and the equality check is only on raw_id not on other fields, so it does make sense for us to add tests where we ensure that if we don't provide the cloud it should be set to the public cloud by default

Reply:
added more tests